### DISCUSSION TITLE
Revisit allowing casks as dependencies for formulas (at least for taps)?

### DISCUSSION THREAD

**Discussion by lgarron** [Homebrew/discussions#5788] [Casks]
### Output of `brew config`

```shell
—
```

### Output of `brew doctor`

```shell
—
```

### Description of issue

I'm finding more and more that I'm unable to write formulas that would properly install on a new system, because they depend on binaries that get installed with casks. For example, just in the last 2 days I've worked with formulas that depend on the following:

- `openscad` (installs `openscad` binary linked from `/Applications/OpenSCAD.app/Contents/MacOS/OpenSCAD`)
- `arq` (installs `arqc` binary linked from `/Applications/Arq.app/Contents/Resources/arqc`)
- `usr-sse2-rdm` (`/Applications/RDM.app/Contents/MacOS/RDM` can be invoked as a commandline binary)

There are also other commonly used binaries installed through casks, such as the `code` binary installed by `visual-studio-code`.

In some other cases this is resolved by including both a formula in `homebrew-core` as well an identically named cask in `homebrew-cask`, such as `handbrake`. However, I get the feeling Homebrew wouldn't want to expand this to casks like the ones I mentioned above. Having separate formulas:

- Is more maintenance work, and there will be a functionality gap unless every cask for an app with a binary also has a corresponding formula for the binary.
- Could introduce a bunch more identically named formulas and casks in the default Homebrew installation, which could cause more problems and confusion than necessary.
- Introduces the possibility that the binary and the app version could get out of sync and therefore install versions with incompatible functionality.
- Might run into issues installing the binaries without also installing the full apps.

As someone who wants to make sure their computer configuration properly expresses dependencies, I would find it really useful if a formula can also depend on a cask, at least in the taps I maintain.

There are some existing discussions about aspects of this, such as https://github.com/orgs/Homebrew/discussions/3163 discussing Docker. @SMillerDev's reply to one of the comments implies that `depends_on cask: "…"` should work for a formula in a tap, but I ever got this to work — it just produces an `Unsupported special dependency :cask` error. So it seems this isn't really robustly supported by the current Homebrew code, and therefore I'm asking here!

### Relevant casks

—

**Comment by p-linnane** [comment]
I am personally not opposed to allowing 3rd party taps to depend on casks. It would very much be an unsupported use case though. Software in Homebrew/core is required to be open-source using an OSI approved license and build from source on our CI, which is why we do not allow cask dependencies. Note that we have things like [`code-cli`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/c/code-cli.rb) which is the VC Code cli component since it fulfills all the requirements for inclusion in Homebrew/core.

**Reply by briceburg** [reply]
As another use-case, we have a private tap for installing a developer facing tool that requires the AWS session manager cask. being able to include this in our formula so we can simply do a `brew install dev-tool` without an additional step would improve the experience.... thanks for considering!! ❤️

**Reply by p-linnane** [reply]
No other maintainers weighed in here, so it's safe to assume that none are interested in making this happen. It would need to be a contribution from the community if they want to see it happen.

**Reply by lgarron** [reply]
> No other maintainers weighed in here, so it's safe to assume that none are interested in making this happen. It would need to be a contribution from the community if they want to see it happen.

Sounds totally fair, but also sounds like no one opposed!

Is this something that could be done by sending a direct PR, or would it require more planning?

**Reply by p-linnane** [reply]
Yes, taking the time to write the code and sending a PR will get you better feedback. We always appreciate PR's and will be able provide additional guidance if you get stuck. We have many carve outs for things for third-party taps already. I can't guarantee 100% that it will get merged, but the odds are much better! 😄

**Comment by gibfahn** [comment]
Just cross-linking to other times this has been discussed: https://github.com/Homebrew/brew/issues/17326, https://github.com/orgs/Homebrew/discussions/3163
